students/shawn_degraw/session08/circle.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+"""
+Class to implement simple Circle
+
+"""
+
+from math import pi
+
+
+class Circle:
+    """
+    Circle class with area and diameter properties
+    numberic protocols added
+
+    """
+
+    def __init__(self, init_radius=0):
+        """ Constructor using radius """
+
+        self.radius = init_radius
+
+    @classmethod
+    def from_diameter(cls, init_diameter=0):
+        """ Contructor using diameter """
+
+        radius = init_diameter / 2
+        return cls(radius)
+
+    @property
+    def diameter(self):
+        return 2 * self.radius
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        return round(pi * self.radius ** 2, 6)
+
+    def __str__(self):
+        return "Circle with radius: {}".format(self.radius)
+
+    def __repr__(self):
+        return "Circle({})".format(self.radius)
+
+    def __add__(self, other):
+        return type(self)(self.radius + other.radius)
```

Comment:
Interesting to see this. I thought `Circle(self.radius + other.radius)` would be just fine and desirable, but it looks like this makes more sense or desirable (unless there's some Python gotcha I don't know about)... Thanks for being inquisitive!